app/routes/press-releases/press-release/edit.js

```diff
@@ -0,0 +1,18 @@
+import Route from '@ember/routing/route';
+import PressReleaseSnapshot from '../../../utils/press-release-snapshot';
+
+export default class PressReleasesPressReleaseEditRoute extends Route {
+  async beforeModel() {
+    const isPublished = (await this.modelFor('press-releases.press-release')).publicationEvent.get('isPublished');
```

Comment:
[`modelFor`](https://api.emberjs.com/ember/release/classes/Route/methods/modelFor?anchor=modelFor) returned geen promise, maar een object. De `await` is dus niet noodzakelijk.

Het ophalen van een relation (vb. `pressRelease.publicationEvent`) in ember-data daarentegen geeft wel een promise terug. Die moet wel ge-`await` worden. Dan wordt de `get()` ineens ook overbodig.

Reply:
ik haal de isPublished niet direct op, want bij een nieuw press release bestaat het niet